skabase/SKABaseDevice/SKABaseDevice.py

```diff
@@ -50,15 +48,18 @@ class SKABaseDevice(with_metaclass(DeviceMeta, Device)):
     # PROTECTED REGION ID(SKABaseDevice.class_variable) ENABLED START #
 
     def _init_syslog(self):
+        self.formatter = logging.Formatter('%(name)s: %(levelname)s %(module)s %(message)r')
         try:
             self.logger = logging.getLogger(__name__)
-            #self.syslogs = SysLogHandler(address='/dev/log', facility='syslog')
+            # self.syslogs = SysLogHandler(address='/dev/log', facility='syslog')
             self.syslogs = SysLogHandler(address='/var/run/rsyslog/dev/log', facility='syslog')
-            self.formatter = logging.Formatter('%(name)s: %(levelname)s %(module)s %(message)r')
             self.syslogs.setFormatter(self.formatter)
             self.logger.addHandler(self.syslogs)
-        except Exception:
-            self.error_stream("Syslog cannot be initialized:")
+        except EnvironmentError as ex:
+            log_file_name = self.__class__.__name__
+            log_file_name += ".log"
+            self.filehandler = WatchedFileHandler(log_file_name, 'a', None, False)
```

Comment:
Sorry but I dont understand "Do we really need to keep a reference...". Do you mean to say the variable 'filehandler' should be addressed without 'self.'? If I am not wrong that will make the variable local to the function and self.xxx makes it class member, isn't it?

Reply:
Yes, I'm suggesting we drop `self.`, so that it is just a local variable, instead of a class variable.  I don't think the developer will need to access it outside of this function.  Obviously the logger will get a reference, and use it for writing out the logs.

Reply:
Ok. Got it. Thanks.

Reply:
As suggested, made the variables local.